omni-nodes/common/src/fake_runtime.rs

```diff
@@ -0,0 +1,202 @@
+//! # Fake Runtime
+//!
+//! A fake runtime that implements (hopefully) all of the runtime apis defined in polkadot-sdk as a
+//! stub. All implementations are fulfilled with `unreachable!()`.
+//!
+//! See [`FakeRuntime`]
+
+#![allow(unused_variables)]
+
+use sp_runtime::traits::Block as BlockT;
+
+/// The fake runtime.
+pub struct FakeRuntime;
+
+/// Some types that we need to fulfill the trait bounds at compile-time, but in runtime they don't
+/// matter at all.
+mod whatever {
+	pub(crate) use frame::runtime::types_common::OpaqueBlock as Block;
+	pub(crate) type Header = <Block as sp_runtime::traits::Block>::Header;
+	pub(crate) type AccountId = sp_runtime::AccountId32;
+	pub(crate) type Nonce = u32;
+	pub(crate) type AuraId = sp_consensus_aura::sr25519::AuthorityId;
+	pub(crate) type BlockNumber = u32;
+	pub(crate) type Balance = u128;
+	pub(crate) type Weight = sp_weights::Weight;
+	pub(crate) type RuntimeCall = ();
+}
+use whatever::*;
+
+sp_api::impl_runtime_apis! {
```

Comment:
This should ideally move to `cumulus_service` now. I think one set of fake runtime impls should be enough.